packages/react-core/src/components/Tabs/Tabs.tsx

```diff
@@ -16,35 +16,52 @@ export enum TabsVariant {
 }
 
 export interface TabsProps extends Omit<React.HTMLProps<HTMLElement | HTMLDivElement>, 'onSelect'> {
-  /** content rendered inside the Tabs Component. */
+  /** Content rendered inside the tabs component. */
   children: React.ReactNode;
-  /** additional classes added to the Tabs */
+  /** Additional classes added to the tabs */
   className?: string;
-  /** the index of the active tab */
+  /** The index of the active tab */
   activeKey?: number | string;
-  /** handle tab selection */
+  /** Callback to handle tab selection */
   onSelect?: (event: React.MouseEvent<HTMLElement, MouseEvent>, eventKey: number | string) => void;
-  /** uniquely identifies the Tabs */
+  /** Uniquely identifies the tabs */
   id?: string;
-  /** enables the filled tab list layout */
+  /** Enables the filled tab list layout */
   isFilled?: boolean;
-  /** enables Secondary Tab styling */
+  /** Enables secondary tab styling */
   isSecondary?: boolean;
-  /** Aria-label for the left Scroll Button */
+  /** Enables box styling to the tab component */
+  isBox?: boolean;
+  /** Enables vertical tab styling */
+  isVertical?: boolean;
+  /** Aria-label for the left scroll button */
   leftScrollAriaLabel?: string;
-  /** Aria-label for the right Scroll Button */
+  /** Aria-label for the right scroll button */
   rightScrollAriaLabel?: string;
-  /** determines what tag is used around the Tabs. Use "nav" to define the Tabs inside a navigation region */
+  /** Determines what tag is used around the tabs. Use "nav" to define the tabs inside a navigation region */
   variant?: 'div' | 'nav';
-  /** provides an accessible label for the Tabs. Labels should be unique for each set of Tabs that are present on a page. When variant is set to nav, this prop should be defined to differentiate the Tabs from other navigation regions on the page. */
+  /** Provides an accessible label for the tabs. Labels should be unique for each set of tabs that are present on a page. When variant is set to nav, this prop should be defined to differentiate the tabs from other navigation regions on the page. */
   'aria-label'?: string;
-  /** waits until the first "enter" transition to mount tab children (add them to the DOM) */
+  /** Waits until the first "enter" transition to mount tab children (add them to the DOM) */
   mountOnEnter?: boolean;
-  /** unmounts tab children (removes them from the DOM) when they are no longer visible */
+  /** Unmounts tab children (removes them from the DOM) when they are no longer visible */
   unmountOnExit?: boolean;
+  /* Modifies the tabs component padding/inset to visually match padding of other adjacent components.*/
+  inset?: 'none' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
```

Comment:
I experimented with our breakpoint modifier API and I think it'd be a good fit and extendable to future modifiers:
```ts
export interface TabBreakpointMod {
  /** The attribute to modify  */
  modifier:
    | 'insetNone'
    | 'insetSm'
    | 'insetMd'
    | 'insetLg'
    | 'insetXl'
    | 'inset-2xl';
  /** The breakpoint at which to apply the modifier */
  breakpoint?: 'sm' | 'md' | 'lg' | 'xl' | '2xl';
}
```
...
```tsx
  /** Array of objects representing the various modifiers to apply to tabs at various breakpoints */
  breakpointMods?: TabBreakpointMod[];
```
...
```
formatBreakpointMods(breakpointMods, styles),
```
...
```tsx
<Tabs
  activeKey={activeTabKey}
  onSelect={this.handleTabClick}
  breakpointMods={[
    { breakpoint: 'md', modifier: 'insetSm' },
    { breakpoint: 'lg', modifier: 'insetLg' },
    { breakpoint: 'xl', modifier: 'inset_2xl' }
  ]}
  isBox={isBox}>
```

Let me know what you think!
```

Reply:
Fair enough.  I see how this allows for a more flexible interface.  But, I do we think if we make this change we should probably go back and update Drawer for consistency.

Reply:
I agree. I think Drawer should use the same API and we should document it in the breaking change release notes.

Reply:
Drawer is still beta.  Do you mind opening the drawer issue.  I will update the Tabs API.

Reply:
I opened #4238